gprofiler/profilers/python.py

```diff
@@ -40,6 +40,7 @@
 from gprofiler.metadata.system_metadata import get_arch
 from gprofiler.profilers.profiler_base import ProfilerBase, ProfilerInterface, SpawningProcessProfilerBase
 from gprofiler.profilers.registry import ProfilerArgument, register_profiler
+from gprofiler.profilers.python_ebpf import PythonEbpfProfiler, PythonEbpfError
```

Comment:
So as I suggested in https://github.com/Granulate/gprofiler/pull/475/files#r973784312, let's refrain from importing it if Windows.

And then, to ensure the name `PythonEbpfProfiler` isn't used, we will:
1. Add logic in `PythonProfiler.__init__` that like the `if get_arch() != "x86_64"` will ensure we're on Linux, otherwise log a warning `PyPerf is supported only on Linux, falling back to py-spy`.
2. This way, `_create_ebpf_profiler` is never called.
3. `_create_ebpf_profiler` itself can be defined in the class `PythonEbpfProfiler` under `if is_windows()`, i.e:
```
class PythonEbpfProfiler(...):
    if is_windows():
    def _create_ebpf_profiler(
            self,
            frequency: int,
            duration: int,
            stop_event: Event,
            storage_dir: str,
            profile_spawned_processes: bool,
            add_versions: bool,
            user_stacks_pages: Optional[int],
        ) -> Optional[PythonEbpfProfiler]:
            ....
    ....
```
its code & definition never runs so it's okay `PythonEbpfProfiler` isn't defined
5. All other branches containing references to names imported from `python_ebpf` are under `if self._ebpf_profiler is not None` which will never be true, so we're protected :+1: 

After these changes, this PR is ready to go.

Reply:
Made requested changes. Please review.

Reply:
Only point 1 remains as commented [here](https://github.com/Granulate/gprofiler/pull/475#discussion_r976991116)